packages/components/src/filters/advanced/number-filter.js

```diff
@@ -0,0 +1,184 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { Component, Fragment } from '@wordpress/element';
+import { SelectControl, TextControl } from '@wordpress/components';
+import { get, find, partial } from 'lodash';
+import interpolateComponents from 'interpolate-components';
+import classnames from 'classnames';
+import { _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import TextControlWithAffixes from '../../text-control-with-affixes';
+import { formatCurrency } from '../../../../currency/src';
+
+class NumberFilter extends Component {
+	getBetweenString() {
+		return _x(
+			'{{moreThan /}} and {{lessThan /}}',
+			'Numerical range inputs arranged on a single line',
+			'wc-admin'
+		);
+	}
+
+	getLegend( filter, config ) {
+		const inputType = get( config, [ 'input', 'type' ], 'number' );
+		const rule = find( config.rules, { value: filter.rule } ) || {};
+		let filterStr = filter.value || '';
+
+		if ( 'between' === rule.value ) {
+			let [ moreThan, lessThan ] = filterStr.split( ',' );
+
+			if ( 'currency' === inputType ) {
+				moreThan = formatCurrency( moreThan );
+				lessThan = formatCurrency( lessThan );
+			}
+
+			filterStr = interpolateComponents( {
+				mixedString: this.getBetweenString(),
+				components: {
+					moreThan: <Fragment>{ moreThan }</Fragment>,
+					lessThan: <Fragment>{ lessThan }</Fragment>,
+				},
+			} );
+		} else if ( 'currency' === inputType ) { // need to format a single input value as currency
+			filterStr = formatCurrency( filterStr );
+		}
+
+		return interpolateComponents( {
+			mixedString: config.labels.title,
+			components: {
+				filter: <span>{ filterStr }</span>,
+				rule: <span>{ rule.label }</span>,
+			},
+		} );
+	}
+
+	getFormControl( type, value, onChange ) {
+		if ( 'currency' === type ) {
+			const currencySymbol = get( wcSettings, [ 'currency', 'symbol' ] );
+			const symbolPosition = get( wcSettings, [ 'currency', 'position' ] );
+
+			return (
+				0 === symbolPosition.indexOf( 'right' )
+				? <TextControlWithAffixes
+					suffix={ <span dangerouslySetInnerHTML={ { __html: currencySymbol } } /> }
+					className="woocommerce-filters-advanced__input-numeric-range"
+					type="number"
+					value={ value }
+					onChange={ onChange }
+				/>
+				: <TextControlWithAffixes
```

Comment:
All 3 of these text controls need labels– You should be able to use the `aria-label` attribute to pass a label directly to the input, which won't display to sighted users.